cpp/tests/test_ide_secir.cpp

```diff
@@ -0,0 +1,229 @@
+    mio::TimeSeries<ScalarType> init(num_transitions);
+
+    // add time points for initialization for transitions and death
+    Vec vec_init(num_transitions);
+    vec_init << 1.0, 0.0, 8.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0;
+    // add initial time point to time series
+    init.add_time_point(-1, vec_init);
+    while (init.get_last_time() < 0) {
+        init.add_time_point(init.get_last_time() + dt, vec_init);
+    }
+
+    // Initialize model.
+    mio::isecir::Model model(std::move(init), dt, N, Dead_before);
+
+    // Set working parameters.
+    model.parameters.set<mio::isecir::TransitionDistributions>(
+        std::vector<mio::isecir::DelayDistribution>(num_transitions, mio::isecir::DelayDistribution()));
+    std::vector<ScalarType> vec_prob((int)mio::isecir::InfectionTransitions::Count, 0.5);
+    vec_prob[Eigen::Index(mio::isecir::InfectionTransitions::SusceptibleToExposed)]        = 1;
+    vec_prob[Eigen::Index(mio::isecir::InfectionTransitions::ExposedToInfectedNoSymptoms)] = 1;
+    model.parameters.set<mio::isecir::TransitionProbabilities>(vec_prob);
+    mio::ContactMatrixGroup contact_matrix               = mio::ContactMatrixGroup(1, 1);
+    contact_matrix[0]                                    = mio::ContactMatrix(Eigen::MatrixXd::Constant(1, 1, 1.));
+    model.parameters.get<mio::isecir::ContactPatterns>() = mio::UncertainContactMatrix(contact_matrix);
+    model.parameters.set<mio::isecir::TransmissionProbabilityOnContact>(0.5);
+    model.parameters.set<mio::isecir::RelativeTransmissionNoSymptoms>(1.0);
+    model.parameters.set<mio::isecir::RiskOfInfectionFromSymptomatic>(1.0);
+
+    // Carry out simulation.
+    mio::TimeSeries<ScalarType> secihurd_simulated    = model.simulate(tmax);
+    mio::TimeSeries<ScalarType> transitions_simulated = model.get_transitions();
+
+    // Define vectors with values from example (calculated by hand, see Overleaf document)
+    Vec secihurd0((int)mio::isecir::InfectionState::Count);
+    Vec secihurd1((int)mio::isecir::InfectionState::Count);
+    Vec transitions1(num_transitions);
+    secihurd0 << 4995, 0.5, 0, 4, 0, 0, 4990.5, 10;
+    secihurd1 << 4994.00020016, 0.49989992, 0.49994996, 0.12498749, 1.03124687, 0.25781172, 4993.45699802, 10.12890586;
+    transitions1 << 0.99979984, 0.99989991, 0.24997498, 0.24997498, 2.06249374, 2.06249374, 0.51562344, 0.51562344,
+        0.12890586, 0.12890586;
+
+    // Compare SECIHURD compartments at times 0 and 1
+    for (Eigen::Index i = 0; i < (Eigen::Index)mio::isecir::InfectionState::Count; i++) {
+        EXPECT_NEAR(secihurd_simulated[0][i], secihurd0[i], 1e-8);
+        EXPECT_NEAR(secihurd_simulated[1][i], secihurd1[i], 1e-8);
+    }
+
+    // Compare transitions at time 1
+    for (Eigen::Index i = 0; i < num_transitions; i++) {
+        EXPECT_NEAR(transitions_simulated[transitions_simulated.get_num_time_points() - 1][i], transitions1[i], 1e-8);
+    }
+}
```

Comment:
We can't implement exactly this test as we need some persons moving either from E to C or from C to I to have an infection dynamic (otherwise there are no new infections and our model cannot deal with this at the moment, this leads to an error in computation). But there is now a test that shows that a different choice for the gammas leads to the same equlibrium (and that the equilibrium is reached sooner if xright is smaller).

Reply:
However, the proportion between Recovered and Dead is not as I expected, we should discuss that.